core/toolchain.go

```diff
@@ -80,26 +80,40 @@ func getToolPath(toolUnqualified string) (toolPath string) {
 	return
 }
 
-// Run the compiler with the -print-file-name option, and return the directory
-// name of the result, or an empty list if a non-existent directory was
-// returned or an error occurred.
-func getFileNameDir(tc toolchain, basename string) (dirs []string) {
+// Run the compiler with the -print-file-name option, and return the result.
+// Check that the file exists. Return an error if the file can't be located.
+func getFileName(tc toolchain, basename string) (fname string, e error) {
 	ccBinary, flags := tc.getCCompiler()
 
 	cmd := exec.Command(ccBinary, utils.NewStringSlice(flags, []string{"-print-file-name=" + basename})...)
 	bytes, err := cmd.Output()
 	if err != nil {
-		fmt.Printf("Error: Couldn't get path for %s: %v\n", basename, err)
+		e = fmt.Errorf("Couldn't get path for %s: %v", basename, err)
 		return
 	}
 
-	fname := strings.TrimSpace(string(bytes))
+	fname = strings.TrimSpace(string(bytes))
+
 	if _, err := os.Stat(fname); os.IsNotExist(err) {
```

Comment:
Did we want to expand this condition, e.g. general "cannot stat" errors, or if it's a directory/otherwise unreadable?

Reply:
D'oh this is existing code. I'm getting mixed up with the compiler runtime commit.